pkg/diskmaker/diskmaker.go

```diff
@@ -180,8 +192,27 @@ func (d *DiskMaker) symLinkDisks(diskConfig *DiskConfig) {
 
 	for storageClass, deviceArray := range deviceMap {
 		for _, deviceNameLocation := range deviceArray {
+			// get PV creation lock which checks for existing symlinks to this device
+			pvLock, pvLocked, existingSymlinks, err := internal.GetPVCreationLock(
+				deviceNameLocation.diskNamePath,
+				d.symlinkLocation,
+			)
+			unlockFunc := func() {
+				err := pvLock.Unlock()
+				if err != nil {
+					klog.Errorf("failed to unlock device: %+v", err)
+				}
+			}
+			defer unlockFunc()
+			if len(existingSymlinks) > 0 { // already claimed, fail silently
+				continue
+			} else if err != nil || !pvLocked { // locking failed for some other reasion
+				klog.Errorf("not symlinking, could not get lock: %v", err)
+				continue
+			}
+
 			symLinkDirPath := path.Join(d.symlinkLocation, storageClass)
-			err := os.MkdirAll(symLinkDirPath, 0755)
+			err = os.MkdirAll(symLinkDirPath, 0755)
```

Comment:
Should we create this directory regardless of error above? Because without that, the code throws:

> E0727 19:27:58.085659       1 discovery.go:201] Error reading directory: open /mnt/local-storage/two: no such file or directory

Reply:
not related to my PR. This is existing functionality. PR scope is already huge.